### DISCUSSION TITLE
Can't manage to make my server public

### DISCUSSION THREAD

**Discussion by alexugthub** [nginx/unit#1464] [Support]
On Ubuntu 24, I can access the server locally (`curl -i 127.0.0.1` does return the index page content) but I cannot access the HTTP protocol and the server's name from outside.
These are my open ports in the firewall (there's no profile for Unit, so I just added 80/tcp):

```
22/tcp                     ALLOW       Anywhere
9090/tcp                   ALLOW       Anywhere
OpenSSH                    ALLOW       Anywhere
80/tcp                     ALLOW       Anywhere
22/tcp (v6)                ALLOW       Anywhere (v6)
9090/tcp (v6)              ALLOW       Anywhere (v6)
OpenSSH (v6)               ALLOW       Anywhere (v6)
80/tcp (v6)                ALLOW       Anywhere (v6)
```

I sent this configuration

```
{
    "listeners": {
        "127.0.0.1:80": {
            "pass": "routes"
        }
    },

    "routes": [
        {
            "action": {
                "share": "/srv/www/public$uri"
            }
        }
    ]
}
```

using `curl -X PUT --data-binary @unit.config.json my.server:9090/config` (I have port 9090 set up for accessing the API behind the router only)

I tried using `*:80` for the listener, but when I try to send it, I get

    "error": "Failed to apply new configuration."

What am I doing wrong?

**Comment by ac000** [comment]
You're using the loopback address (127.0.0.1), that means it is only accessible from _that_ host.

You need to use either the **INADDR_ANY** IPv4 address `0.0.0.0` and or the **in6any_addr** IPv6 address `::` or an assigned IP address of the machine.

**Reply by alexugthub** [reply]
I still get the same error. Like I said, even when using a wildcard

```
{
    "listeners": {
        "*:80": {
            "pass": "routes"
        }
    },

    "routes": [
        {
            "action": {
                "share": "/srv/www/public$uri"
            }
        }
    ]
}
```

I still get `"error": "Failed to apply new configuration."`. It only works when using `127.0.0.1:80`

**Comment by alexugthub** [comment]
I checked for what uses port 80 (I knew Apache was stopped and nginx uninstalled)

```
tcp        0      0 127.0.0.1:80            0.0.0.0:*               LISTEN      150750/unit: router
```
I see `150750/unit: router`, I tried to kill it but it doesn't work. In any case, I used a different port for now

**Comment by ac000** [comment]
You likely needed to delete the Unit config

E.g.

```
$ curl -X DELETE --unix-socket /opt/unit/control.unit.sock http://localhost/config/
```

Before reconfiguring to listen on all interfaces (you likely had a bind(2) error in your unit.log about "address already in use").
